### PR TITLE
fix: faucet broken application 2

### PR DESCRIPTION
**Pull request by GabrielPrediger**

### CODE REVIEW

File: source/state/vault/index.ts [hunk 1]
```diff
@@ -308,13 +308,23 @@ const VaultState = createSlice({
     },
     setFaucetModalState: (
       state: IVaultState,
-      action: PayloadAction<number>
+      action: PayloadAction<{ chainId: number; isFirstTime?: boolean }>
```
Comment (lucaslovato) [inline source/state/vault/index.ts:311]: with the migration isFirstTime is not needed anymore?

File: source/state/vault/index.ts [hunk 2]
```diff
@@ -308,13 +308,23 @@ const VaultState = createSlice({
...
     ) => {
+      const { chainId, isFirstTime } = action.payload;
       if (state.isBitcoinBased) {
         return;
       }
 
-      state.faucetModal[action.payload] = false;
+      if (isFirstTime && !state.isBitcoinBased) {
```
Comment (lucaslovato) [inline source/state/vault/index.ts:318]: this will be replaced with the migration, right?

File: source/state/vault/index.ts (hunk 1 above)
Comment (GabrielPrediger) [reply source/state/vault/index.ts:311]: are separate cases, isfirstime will still remain necessary as we need to know if the user has not yet interacted with the modals in the faucet networks

File: source/state/vault/index.ts (hunk 2 above)
Comment (GabrielPrediger) [reply source/state/vault/index.ts:318]: probably yes, we can discuss whether it would still be interesting to leave it to have an extra "protection"

**Pull request closed without merging**